browser/app/js/objects/actions.js

```diff
@@ -35,34 +46,35 @@ export const SET_SHARE_OBJECT = "objects/SET_SHARE_OBJECT"
 export const CHECKED_LIST_ADD = "objects/CHECKED_LIST_ADD"
 export const CHECKED_LIST_REMOVE = "objects/CHECKED_LIST_REMOVE"
 export const CHECKED_LIST_RESET = "objects/CHECKED_LIST_RESET"
+export const SET_LIST_LOADING = "objects/SET_LIST_LOADING"
 
-export const setList = (objects, marker, isTruncated) => ({
+export const setList = (objects) => ({
   type: SET_LIST,
   objects,
-  marker,
-  isTruncated
 })
 
 export const resetList = () => ({
   type: RESET_LIST
 })
 
-export const appendList = (objects, marker, isTruncated) => ({
-  type: APPEND_LIST,
-  objects,
-  marker,
-  isTruncated
+export const setListLoading = listLoading => ({
+  type: SET_LIST_LOADING,
+  listLoading
 })
 
-export const fetchObjects = append => {
+export const fetchObjects = () => {
   return function(dispatch, getState) {
-    const {buckets: {currentBucket}, objects: {currentPrefix, marker}} = getState()
+    dispatch(resetList())
+    dispatch(setListLoading(true))
```

Comment:
When the user doesn't have any buckets, the loader keeps on spinning as follows:

![Screen Shot 2019-06-20 at 6 28 07 PM](https://user-images.githubusercontent.com/27314853/59891274-5c4f1500-9389-11e9-84fc-7d9872c75ff6.png)

This is probably caused by this line not being inside the `if (currentBucket)` condition, therefore causing the `listLoading` value to be true even before there is a list to load.

Reply:
Done